tests/zfs-tests/tests/functional/cli_root/zfs_mount/zfs_mount_006_pos.ksh

```diff
@@ -94,17 +96,31 @@ log_must $ZFS create $TESTPOOL/$TESTFS1
 unmounted $TESTPOOL/$TESTFS1 || \
 	log_must force_unmount $TESTPOOL/$TESTFS1
 
-while [[ -n $mtpt ]] ; do
-	(( depth == MAXDEPTH )) && \
+while (( depth >= 0 )) ; do
+	if (( depth == MAXDEPTH )) ; then
 		log_note "Verify that '$ZFS $mountcmd <filesystem>' " \
 		"which mountpoint be the identical of an existing one " \
-		"will fail with return code 1."
-
-	log_must $ZFS set mountpoint=$mtpt $TESTPOOL/$TESTFS1
-	log_mustnot $ZFS $mountcmd $TESTPOOL/$TESTFS1
-
-	unmounted $TESTPOOL/$TESTFS1 || \
-		log_fail "Filesystem $TESTPOOL/$TESTFS1 is mounted."
+		"will fail with return code 1, but it will succeed on linux."
+		if is_linux; then
+			log_must $ZFS set mountpoint=$mtpt $TESTPOOL/$TESTFS1
+			log_must $ZFS $mountcmd $TESTPOOL/$TESTFS1
+			
+			mounted $TESTPOOL/$TESTFS1 || \
+				log_fail "Filesystem $TESTPOOL/$TESTFS1 is unmounted."
+		else
+			log_must $ZFS set mountpoint=$mtpt $TESTPOOL/$TESTFS1
+			log_mustnot $ZFS $mountcmd $TESTPOOL/$TESTFS1
+			
+			unmounted $TESTPOOL/$TESTFS1 || \
+				log_fail "Filesystem $TESTPOOL/$TESTFS1 is mounted."
+		fi
+	else
+		$ZFS set mountpoint=$mtpt $TESTPOOL/$TESTFS1
```

Comment:
/IDE0/ztest/zfs-tests/bin # ./zfs list
NAME                                          USED  AVAIL  REFER  MOUNTPOINT
vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/3    29K   153G    29K  /IDE0/liuhua/mntttt
vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/4    29K   153G    29K  /IDE0/liuhua/mntttt
/IDE0/ztest/zfs-tests/bin # ./zfs mount vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/4
/IDE0/ztest/zfs-tests/bin # mount 
tmpfs on /dev/shm type tmpfs (rw,relatime)
vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/4 on /IDE0/liuhua/mntttt type zfs (rw,relatime,xattr,noacl)
/IDE0/ztest/zfs-tests/bin # **./zfs set mountpoint=/IDE0/liuhua vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/4**
cannot mount '/IDE0/liuhua': directory is not empty
property may be set but unable to remount filesystem
/IDE0/ztest/zfs-tests/bin # **./zfs set mountpoint=/IDE0 vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/4**       
/IDE0/ztest/zfs-tests/bin # ./zfs list
NAME                                          USED  AVAIL  REFER  MOUNTPOINT
vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/3    29K   153G    29K  /IDE0/liuhua/mntttt
vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/4    29K   153G    29K  **/IDE0**
/IDE0/ztest/zfs-tests/bin # mount
tmpfs on /dev/shm type tmpfs (rw,relatime)
/IDE0/ztest/zfs-tests/bin # ./zfs mount vd-a-8423E2A3-9C85-6115-1C04-AC3CDC5DA29B/4
cannot mount '/IDE0': directory is not empty

**At first,  the error occurred when "zfs set mountpoint". But I try it again, the mountpoint has changed. So I don't add "log_must"**